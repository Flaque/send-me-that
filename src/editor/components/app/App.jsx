import React from 'react';
import CodeEditor from '../codeEditor/CodeEditor.jsx';
import Output from '../output/Output.jsx';
import ButtonGroup from '../buttonGroup/ButtonGroup.jsx';

class App extends React.Component {

  constructor(props) {
    super(props)
    this.state = { code: '', runnableCode: '' }
    this.onRun = this.onRun.bind(this);
    this.onInsert = this.onInsert.bind(this);
    this.onCodeChange = this.onCodeChange.bind(this);
  }

  onRun() {
    this.setState((prevState) => ({
      runnableCode : prevState.code
    }))
  }

  onInsert() {
    console.log("Insert button pressed");
  }

  onCodeChange(value) {
    this.setState({code: value})
  }

  render() {
    return (
      <div className="app card">
        <CodeEditor onCodeChange={this.onCodeChange} />
        <Output code={this.state.runnableCode}/>
        <ButtonGroup onRun={this.onRun} onInsert={this.onInsert}/>
      </div>
    )
  }
}

export default App;
