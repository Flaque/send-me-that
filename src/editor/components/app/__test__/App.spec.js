import React from 'react';
import App from '../App.jsx';
import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-addons-test-utils';

test('Test that App is still the same', () => {

  // We're currently using the shallow renderer to avoid
  // issues with refs in the CodeEditor module.
  // See: https://github.com/facebook/react/issues/7148
  const shallowRenderer = ReactTestUtils.createRenderer();
  const result = shallowRenderer.render(<App />);
  expect(result).toMatchSnapshot();
})
