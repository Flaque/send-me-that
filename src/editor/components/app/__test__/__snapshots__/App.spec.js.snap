exports[`test Test that App is still the same 1`] = `
<div
  className="app">
  <CodeEditor />
</div>
`;
